packages/react-native/ReactCommon/react/renderer/components/textinput/iostextinput/react/renderer/components/iostextinput/TextInputEventEmitter.cpp

```diff
@@ -167,4 +169,39 @@ void TextInputEventEmitter::dispatchTextInputContentSizeChangeEvent(
       priority);
 }
 
+#if TARGET_OS_OSX // [macOS
+void TextInputEventEmitter::onAutoCorrectChange(OnAutoCorrectChange event) const {
+  dispatchEvent("autoCorrectChange", [event=std::move(event)](jsi::Runtime &runtime) {
+    auto payload = jsi::Object(runtime);
+    payload.setProperty(runtime, "enabled", event.enabled);
+    return payload;
+  });
+}
+
+void TextInputEventEmitter::onSpellCheckChange(OnSpellCheckChange event) const {
+  dispatchEvent("spellCheckChange", [event=std::move(event)](jsi::Runtime &runtime) {
+    auto payload = jsi::Object(runtime);
+    payload.setProperty(runtime, "enabled", event.enabled);
+    return payload;
+  });
+}
+
+void TextInputEventEmitter::onGrammarCheckChange(OnGrammarCheckChange event) const {
+  dispatchEvent("grammarCheckChange", [event=std::move(event)](jsi::Runtime &runtime) {
+    auto payload = jsi::Object(runtime);
+    payload.setProperty(runtime, "enabled", event.enabled);
+    return payload;
+  });
+}
+
+void TextInputEventEmitter::onPaste(PasteEvent const &pasteEvent) const {
+  dispatchEvent("paste", [pasteEvent](jsi::Runtime &runtime) {
+    auto payload = jsi::Object(runtime);
+    auto dataTransfer= dataTransferPayload(runtime, pasteEvent.dataTransferItems);
+    payload.setProperty(runtime, "dataTransfer", dataTransfer);
+    return payload;
+  });
+}
+#endif // macOS]
+
```

Comment:
These platform specific files were refactored in core. If we rebase on 0.74, I can move them to their correct folders.
https://github.com/facebook/react-native/pull/41600
`packages/react-native/ReactCommon/react/renderer/components/textinput/platform/macos/react/renderer/components/iostextinput/...`